src/extra/widgets/animbtn/lv_animbtn.h

```diff
@@ -0,0 +1,143 @@
+/**
+ * @file lv_animbtn.h
+ *
+ */
+
+#ifndef LV_ANIMBTN_H
+#define LV_ANIMBTN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../widgets/lv_img.h"
+
+#if LV_USE_ANIMBTN != 0
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+typedef enum {
+    LV_ANIMBTN_STATE_RELEASED = 1,          /* 0001 */
+    LV_ANIMBTN_STATE_PRESSED  = 2,          /* 0010 */
+    LV_ANIMBTN_STATE_DISABLED = 3,          /* 0100 */
+    LV_ANIMBTN_STATE_CHECKED_RELEASED = 4,  /* 1001 */
+    LV_ANIMBTN_STATE_CHECKED_PRESSED  = 5,  /* 1010 */
+    LV_ANIMBTN_STATE_CHECKED_DISABLED = 6,  /* 1100 */
+    _LV_ANIMBTN_STATE_NUM = 6,
+} lv_animbtn_state_t;
+
+
+typedef enum {
+    LV_ANIMBTN_CTRL_FORWARD  = LV_IMG_CTRL_FORWARD,
```

Comment:
Can't we use `LV_IMG_CTRL_...` directly?

Reply:
No, I don't think so. There's only a few control in IMG that can be used in ANIMBTN, so if I remove this enum, devs will wonder why some controls don't work. I think it's better to only list those that are expected to work here and have them named differently.

Reply:
Fair enough.